chore(curriculum): add working with colors transcripts (#58479)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lecture-working-with-colors-in-css/672bc523324694be91d90d96.md

Saturation refers to the intensity or purity of the color. It is measured as a percentage from `0%` (a completely desaturated, grayscale color) to `100%` (the most vibrant form of that color). A color with `100%` saturation is fully vivid, while a color with `0%` saturation appears as a shade of gray.

Lightness determines how light or dark the color is, again measured as a percentage. A lightness value of `0%` will produce black, `50%` will give you the normal tone of the hue, and `100%` will result in white.

In CSS, the `hsl()` function is used to define colors using the HSL color model. Here is the basic syntax:

```css
element {
  color: hsl(hue, saturation, lightness);
}
```

Let's break this down with an example:

```css
p {
  color: hsl(120, 100%, 50%);
}
```

In this case, the hue is `120` degrees, which corresponds to green. The saturation is `100%`, so the green will be fully vivid. The lightness is `50%`, so it's at its normal tone — neither too dark nor too light. As a result, the text color of the paragraph will be a bright green.

> 

Reviewer: ```suggestion
```

Watch the lecture video and answer the questions below.

# --transcript--

> What is the HSL color model, and how does the HSL function work in CSS?

Reviewer 2: ```suggestion
What is the HSL color model, and how does the `hsl()` function work in CSS?
```

The HSL color model represents colors in a way that is more in line with how humans perceive color. In this lecture, we will explore what the HSL color model is and how the `hsl()` function is used in CSS.

The hue is the color type, represented as an angle on the color wheel, which ranges from `0` to `360` degrees. For example, `0` degrees represents red, `120` degrees is green, and `240` degrees is blue. By adjusting the hue, you shift through different colors on the spectrum.

Saturation refers to the intensity or purity of the color. It is measured as a percentage from `0%` (a completely desaturated, grayscale color) to `100%` (the most vibrant form of that color). A color with `100%` saturation is fully vivid, while a color with `0%` saturation appears as a shade of gray.

Lightness determines how light or dark the color is, again measured as a percentage. A lightness value of `0%` will produce black, `50%` will give you the normal tone of the hue, and `100%` will result in white.

In CSS, the `hsl()` function is used to define colors using the HSL color model. Here is the basic syntax:

```css
element {
  color: hsl(hue, saturation, lightness);
}
```

> Let’s break this down with an example:

Reviewer 2: ```suggestion
Let's break this down with an example:
```

```css
p {
  color: hsl(120, 100%, 50%);
}
```

> In this case, the hue is `120` degrees, which corresponds to green. The saturation is `100%`, so the green will be fully vivid. The lightness is `50%`, so it’s at its normal tone — neither too dark nor too light. As a result, the text color of the paragraph will be a bright green.

Reviewer 2: ```suggestion
In this case, the hue is `120` degrees, which corresponds to green. The saturation is `100%`, so the green will be fully vivid. The lightness is `50%`, so it's at its normal tone — neither too dark nor too light. As a result, the text color of the paragraph will be a bright green.
```

One of the main advantages of the HSL color model is its intuitive nature. It makes it easy to adjust a color’s vibrancy or lightness by tweaking the saturation and lightness values without having to alter the core color (hue). 

For instance, if you want to create different shades or tints of the same color, you can simply adjust the lightness value.

```css
div.light {
  background-color: hsl(240, 100%, 80%);
}

div.dark {
  background-color: hsl(240, 100%, 20%);
}
```

> Here, both div elements are using the same hue (`240` degrees, which is blue), but one has a lightness of `80%` (a lighter shade of blue), and the other has a lightness of `20%` (a darker shade of blue).

Reviewer 3: ```suggestion
Here, both `div` elements are using the same hue (`240` degrees, which is blue), but one has a lightness of `80%` (a lighter shade of blue), and the other has a lightness of `20%` (a darker shade of blue).
```
